services/src/main/java/org/exoplatform/news/search/NewsIndexingServiceConnector.java

```diff
@@ -0,0 +1,177 @@
+  private static final long     serialVersionUID = 1L;
+
+  public static final String    TYPE             = "news";
+
+  private static final Log      LOG              = ExoLogger.getLogger(NewsIndexingServiceConnector.class);
+
+  private final NewsService     newsService;
+
+  private final IdentityManager identityManager;
+
+  public NewsIndexingServiceConnector(IdentityManager identityManager, InitParams initParams, NewsService newsService) {
+    super(initParams);
+    this.newsService = newsService;
+    this.identityManager = identityManager;
+  }
+
+  @Override
+  public String getType() {
+    return TYPE;
+  }
+
+  @Override
+  public Document create(String id) {
+    return getDocument(id);
+  }
+
+  @Override
+  public Document update(String id) {
+    return getDocument(id);
+  }
+
+  @Override
+  public List<String> getAllIds(int offset, int limit) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean canReindex() {
+    return false;
+  }
+
+  private Document getDocument(String id) {
+    if (StringUtils.isBlank(id)) {
+      throw new IllegalArgumentException("id is mandatory");
+    }
+    LOG.debug("Index document for news id={}", id);
+    Document document = new Document();
+    News news = null;
+    String spaceId = null;
+    SessionProvider systemProvider = SessionProvider.createSystemProvider();
+    SessionProviderService sessionProviderService = CommonsUtils.getService(SessionProviderService.class);
+    sessionProviderService.setSessionProvider(null, systemProvider);
+    try {
+      news = newsService.getNewsById(id);
+    } catch (Exception e) {
+      e.printStackTrace();
+    } finally {
+      sessionProviderService.removeSessionProvider(null);
+    }
+    if (news == null) {
+      throw new IllegalStateException("news with id '" + id + "' is mandatory");
+    }
+    Map<String, String> fields = new HashMap<>();
+    fields.put("id", news.getId().replace("comment", ""));
+
+    fields.put("title", news.getTitle());
+
+    String body = news.getBody();
+    if (StringUtils.isBlank(body)) {
+      body = news.getTitle();
+    }
+    fields.put("body", body);
+
+    if (StringUtils.isNotBlank(news.getAuthor())) {
+      fields.put("posterId", news.getAuthor());
+      Identity posterIdentity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, news.getAuthor());
+      if (posterIdentity != null && posterIdentity.getProfile() != null
+          && StringUtils.isNotBlank(posterIdentity.getProfile().getFullName())) {
+        fields.put("posterName", posterIdentity.getProfile().getFullName());
+      }
+    }
+    spaceId = news.getSpaceId();
+    if (spaceId != null) {
+      fields.put("spaceDisplayName", String.valueOf(news.getSpaceDisplayName()));
+    }
+
+    String newsActivityId = news.getActivities();
+    if (newsActivityId != null) {
+      fields.put("newsActivityId", newsActivityId.split(":")[1].substring(0, 1));
```

Comment:
we should get the news activity id as here https://github.com/exoplatform/news/blob/develop/services/src/main/java/org/exoplatform/news/NewsServiceImpl.java#L665